packages/fast-components-react-base/src/text-field/text-field.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "../foundation";
+import {
+    ITextFieldHandledProps,
+    ITextFieldManagedClasses,
+    ITextFieldUnhandledProps,
+    TextFieldType
+} from "./text-field.props";
+import { IManagedClasses, ITextFieldClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/* tslint:disable-next-line */
+class TextField extends Foundation<ITextFieldHandledProps & IManagedClasses<ITextFieldClassNameContract>, React.HTMLAttributes<HTMLInputElement>, {}> {
+    protected handledProps: HandledProps<ITextFieldHandledProps & IManagedClasses<ITextFieldClassNameContract>> = {
+        disabled: void 0,
+        placeholder: void 0,
+        managedClasses: void 0,
+        type: void 0,
+        value: void 0
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLInputElement> {
+        return (
+            <input
+                {...this.unhandledProps()}
+                {...this.generateHandledAttributes()}
+                className={this.generateClassNames()}
+            />
+        );
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        return super.generateClassNames(get(this.props, "managedClasses.textField"));
+    }
+
+    /**
+     * Generates handled attributes based on props
+     */
+    private generateHandledAttributes(): Partial<ITextFieldHandledProps> {
```

Comment:
We are trying to get away from the `generateAttributes` method for a few reasons. The props can go directly on the render function.

Reply:
See checkbox for examples